global/scripts/mppmodecheck.shlib

```diff
@@ -0,0 +1,56 @@
+# Functions for checking compliance with selected MPP mode for HCP Pipelines
+
+# --- Import log.shlib 
+
+if [[ "$(type -t log_Err_Abort)" != "function" ]]
+then
+    source "$HCPPIPEDIR/global/scripts/log.shlib" "$@"
+fi
+
+
+check_mpp_compliance()
+{
+
+    # --- Define variables
+
+    local MPPMode="$1"
+    local Compliance="$2"
+    local ComplianceMsg="$3"
+    local ComplianceWarn="$4"
+
+    # --- Initial information
+
+    if [ "${MPPMode}" = "LegacyStyleData" ]; then
+      log_Msg "LegacyStyleData mode requested."
+      log_Msg "NOTICE: You are using a mode that enables processing of acquisitions that do not"
+      log_Msg "        conform to the HCP specification as described in Glasser et al. (2013)!"
+      log_Msg "        Be aware that if the HCP requirements are not met, the level of data quality"
+      log_Msg "        can not be guaranteed and the Glasser et al. (2013) paper should not be used"
+      log_Msg "        in support of this workflow. A manuscript with comprehensive evaluation for"
+      log_Msg "        the Extended MPP workflow is in active preparation and should be appropriately"
+      log_Msg "        cited when published."
+      log_Msg "        "
+    fi
+
+    # -- Final evaluation
+
+    if [ "${MPPMode}" = "LegacyStyleData" ]; then
+      if [ "${Compliance}" = "HCPStyleData" ]; then
+        log_Warn "All conditions for the use of --mpp-mode=HCPStyleData are met. Consider using that mode (or equivalently, simply leave --mpp-mode empty) instead of --mpp-mode=LegacyStyleData."
+      else
+        log_Msg "The following LegacyStyleData settings were requested:${ComplianceMsg}"
+      fi
+      if [ ! -z "${ComplianceWarn}" ] ; then
+        log_Warn "${ComplianceWarn}"
```

Comment:
I'm guessing this is for the warnings of the nature "freesurfer registration is not as good as MSMSulc, please consider changing"?  Putting it here means that if the script will abort because of a compliance check, it won't print the warning about the requested option.

What happens if there are multiple options that we want warnings on in one script, how do you plan to separate them?  I think just putting the log_Warn directly into the pipeline script is the simplest solution.

@glasserm if someone asks for FS registration and doesn't specify legacy mode, do you want it to show the warning about *why* we recommend not using FS, in addition to aborting?

Reply:
I was looking at ways to combine warnings, but found nothing robust. If you have a suggestion, how to combine and pass them in a single variable and then split and loop through them, it would be most welcome!

Reply:
My suggestion is to never combine them in the first place, it is acceptable, and possibly preferable, to have these warnings unconditional (except for whether the setting was specified, of course), and in the pipeline scripts themselves.

Bash does not have a clean way of passing an array of strings as an argument, as far as I am aware.  Since these are sourced functions, and thus running in the "workspace" of the pipeline script, there are tricks that can be used, but I think it is better to just put the log_Warn directly into the relevant argument check in the main script.

For reference only, within the context of one script and its functions, a global array variable is clean for storing arrays of strings:

```
myarray=()
...
myarray+=("somevar is $somevar")
myarray+=("the quick brown fox")
```

Reply:
Ok. I'll pull the warnings into individual scripts.

Reply:
Yes lets print both the notice and the warning.